src/Device/Driver/XCVario.cpp

```diff
@@ -158,6 +160,23 @@ XVCDevice::ParseNMEA(const char *String, NMEAInfo &info)
     return false;
 }
 
+// For documentation refer to chapter 10.1.3 Device Driver/XCVario in mulilingual handbook: https://xcvario.de/handbuch
+
+bool 
+XVCDevice::PutQNH(const AtmosphericPressure &pres, OperationEnvironment &env)
+{
+  /* the XCVario understands "!g,q<NNNN>" command for QNH updates with recent builds */
+  char buffer[32];
+  unsigned qnh = uround(pres.GetPascal()/100);
+  int msg_len = sprintf(buffer,"!g,q%u\r", std::min(qnh,(unsigned)2000));
+  if ( msg_len > 0 ) {
```

Comment:
What's the point of this check?

Reply:
Avoid to send once formatting to buffer fails.

Reply:
Under what circumstances will this fail?

Reply:
I see the point, we can remove it, see also my other comment, okay ?

Reply:
Then lets rely on sprintf. No problem, remove the extra check.